workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/AndroidRenderWorkflow.kt

```diff
@@ -0,0 +1,121 @@
+ *        workflow = HelloWorkflow,
+ *        scope = this.viewModelScope,
+ *        savedStateHandle = state,
+ *        interceptors = listOf(SimpleLoggingWorkflowInterceptor())
+ *      )
+ *    }
+ */
+@WorkflowUiExperimentalApi
+@OptIn(ExperimentalCoroutinesApi::class, ExperimentalWorkflowApi::class)
+public fun <OutputT, RenderingT> renderWorkflowIn(
+  workflow: Workflow<Unit, OutputT, RenderingT>,
+  scope: CoroutineScope,
+  savedStateHandle: SavedStateHandle? = null,
+  interceptors: List<WorkflowInterceptor> = emptyList(),
+  onOutput: suspend (OutputT) -> Unit = {}
+): StateFlow<RenderingT> {
+  return renderWorkflowIn(
+    workflow = workflow,
+    scope = scope,
+    props = MutableStateFlow(Unit),
+    savedStateHandle = savedStateHandle,
+    interceptors = interceptors,
+    onOutput = onOutput
+  )
+}
+/**
+ * An Android `ViewModel`-friendly wrapper for [com.squareup.workflow1.renderWorkflowIn],
+ * for use with a [workflow] that requires input ([props]) to run.
+ *
+ * For example, for a workflow that uses [android.content.Intent] as its `PropsT` type,
+ * you could do something like this:
+ *
+ *    @OptIn(WorkflowUiExperimentalApi::class)
+ *    class HelloWorkflowActivity : AppCompatActivity() {
+ *      private val model: HelloViewModel by viewModels()
+ *
+ *      override fun onCreate(savedInstanceState: Bundle?) {
+ *        super.onCreate(savedInstanceState)
+ *
+ *        model.props.value = intent
+ *        setContentView(
+ *          WorkflowLayout(this).apply { start(model.renderings) }
+ *        )
+ *      }
+ *
+ *      override fun onNewIntent(intent: Intent) {
+ *        super.onNewIntent(intent)
+ *        model.props.value = intent
+ *      }
+ *    }
+ *
+ *    class HelloViewModel(state: SavedStateHandle) : ViewModel() {
+ *      val props = MutableStateFlow(Intent())
+ *
+ *      @OptIn(WorkflowUiExperimentalApi::class)
+ *      val renderings: StateFlow<HelloRendering> = renderWorkflowIn(
+ *        workflow = HelloWorkflow,
+ *        scope = this.viewModelScope,
+ *        savedStateHandle = state,
+ *        interceptors = listOf(SimpleLoggingWorkflowInterceptor()),
+ *        props = props
+ *      )
+ *    }
+ */
+@WorkflowUiExperimentalApi
+@OptIn(ExperimentalCoroutinesApi::class, ExperimentalWorkflowApi::class)
+public fun <PropsT, OutputT, RenderingT> renderWorkflowIn(
+  workflow: Workflow<PropsT, OutputT, RenderingT>,
+  scope: CoroutineScope,
+  props: StateFlow<PropsT>,
+  savedStateHandle: SavedStateHandle? = null,
+  interceptors: List<WorkflowInterceptor> = emptyList(),
+  onOutput: suspend (OutputT) -> Unit = {}
+): StateFlow<RenderingT> {
+  val restoredSnap = savedStateHandle?.get<PickledWorkflow>(KEY)?.snapshot
+  val renderingsAndSnapshots = renderWorkflowIn(
+    workflow, scope, props, restoredSnap, interceptors, onOutput
+  )
+
+  return renderingsAndSnapshots
+    .onEach { savedStateHandle?.set(KEY, PickledWorkflow(it.snapshot)) }
```

Comment:
lol @`PickledWorkflow` - do we already use this concept in the examples?

Reply:
No, it's internal, and I intend it to stay that way.